aws-rds-optiongroup/src/main/java/software/amazon/rds/optiongroup/CreateHandler.java

```diff
@@ -49,31 +61,39 @@ protected ProgressEvent<ResourceModel, CallbackContext> handleRequest(
                     }
                     return ProgressEvent.progress(model, progress.getCallbackContext());
                 })
-                .then(progress -> proxy.initiate("rds::create-option-group", proxyClient, progress.getResourceModel(), progress.getCallbackContext())
-                        .translateToServiceRequest(model -> Translator.createOptionGroupRequest(
-                                model,
-                                mergeMaps(
-                                        request.getSystemTags(),
-                                        request.getDesiredResourceTags()
-                                )
-                        ))
-                        .backoffDelay(config.getBackoff())
-                        .makeServiceCall((createRequest, proxyInvocation) -> proxyInvocation.injectCredentialsAndInvokeV2(
-                                createRequest,
-                                proxyInvocation.client()::createOptionGroup
-                        ))
-                        .handleError((createRequest, exception, client, resourceModel, ctx) -> Commons.handleException(
-                                ProgressEvent.progress(resourceModel, ctx),
-                                exception,
-                                DEFAULT_OPTION_GROUP_ERROR_RULE_SET
-                        ))
-                        .progress())
+                .then(progress -> createDbClusterParameterGroup(proxy, proxyClient, progress, systemTags))
+                .then(progress -> updateTags(proxy, proxyClient, progress, Tagging.TagSet.emptySet(), extraTags))
                 .then(progress -> {
                     if (CollectionUtils.isNullOrEmpty(progress.getResourceModel().getOptionConfigurations())) {
                         return progress;
                     }
                     return updateOptionGroup(proxy, proxyClient, progress);
                 })
+                .then(progress -> {
+                    return progress;
+                })
                 .then(progress -> new ReadHandler().handleRequest(proxy, request, callbackContext, proxyClient, logger));
     }
+
+    private ProgressEvent<ResourceModel, CallbackContext> createDbClusterParameterGroup(final AmazonWebServicesClientProxy proxy,
+                                                                                        final ProxyClient<RdsClient> proxyClient,
+                                                                                        final ProgressEvent<ResourceModel, CallbackContext> progress,
+                                                                                        final Tagging.TagSet tags) {
+        return proxy.initiate("rds::create-option-group", proxyClient, progress.getResourceModel(), progress.getCallbackContext())
```

Comment:
I believe it needs to be handled in a 2-tries way: 1st with all the tags available and the second with system tags only. Please see this ref as an example: https://github.com/aws-cloudformation/aws-cloudformation-resource-providers-rds/blob/master/aws-rds-dbparametergroup/src/main/java/software/amazon/rds/dbparametergroup/CreateHandler.java#L58-L71

Reply:
If I understand correctly, the 2-tries way would be to address the IAM role with tags issue? As addressed in this PR: https://github.com/aws-cloudformation/aws-cloudformation-resource-providers-rds/pull/188.

Do you think I should do this in this PR or 2 separate PRs?

Reply:
When I implemented soft failing before, We did not know about IAM issue. THat's why we have 2 PRs (soft fail, safe create). Now, We know this approach is not valid any more without safe create that Oleg mentioned. We can not merge code that we sure it is not covering all cases(having a bug).

Reply:
I've added the IAM softfail now